### PR TITLE
Unit Test - Group information screen

### PR DESCRIPTION
**Pull request by at-ngoctran**
#### Related issues (Required)
- Task #199 

#### Description (Required)
- [x] Create unit test for group information screen

### CODE REVIEW

**Comment by at-vinhhuynh** [comment]
@at-ngoctran 
This PR have more code that merged from other PR.
Please re-create a PR again.
thank you!

File: app/src/main/java/vn/asiantech/way/ui/group/info/GroupInfoViewModel.kt [hunk 1]
```diff
@@ -10,8 +10,9 @@ import vn.asiantech.way.data.source.GroupRepository
  * Copyright © 2017 Asian Tech Co., Ltd.
  * Created by cuongcaov on 05/12/2017
  */
-class GroupInfoViewModel {
-    private val groupRepository = GroupRepository()
+class GroupInfoViewModel(private val groupRepository: GroupRepository) {
```
Comment (at-vinhhuynh) [inline app/src/main/java/vn/asiantech/way/ui/group/info/GroupInfoViewModel.kt]: Please remove a java doc

File: app/src/test/java/vn/asiantech/way/group/info/GroupInfoViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        MockitoAnnotations.initMocks(this)
+        viewModel = GroupInfoViewModel(groupRepository)
+    }
+
+    @Test
+    fun `Given a group Id - When call get group info  - Then return a group`() {
+        /* Given */
+        val test = TestObserver<Group>()
+        val groupId = "groupId"
+        val group = Group(id = "id", name = "name", token = "token", ownerId = "ownerId",
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/group/info/GroupInfoViewModelTest.kt]: change this line to `val group = Group("id", "name", "token", "ownerId",
                "createAt", "modifiedAt")`

File: app/src/test/java/vn/asiantech/way/group/info/GroupInfoViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    @Test
+    fun `Given a userId - When call leave group - Then return an user `() {
+        /* Given */
+        val test = TestObserver<User>()
+        val userId = "userId"
+        val user = User()
+        `when`(groupRepository.removeUserFromGroup(TestUtil.any())).thenReturn(Single.just(user))
+
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/group/info/GroupInfoViewModelTest.kt]: remove this line

File: app/src/test/java/vn/asiantech/way/group/info/GroupInfoViewModelTest.kt [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        /* When */
+        viewModel.leaveGroup(userId).subscribe(test)
+
+        /* Then */
+        test.assertValue { it == user }
+    }
+
+    @Test
+    fun `Given a groupId and newOwnerId - When call change group owner  - Then return true `() {
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/group/info/GroupInfoViewModelTest.kt]: `groupId ` change to `group id ` `newOwnerId` change to `new owner id`
also check on method name below

File: app/src/test/java/vn/asiantech/way/group/info/GroupInfoViewModelTest.kt [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+        /* When */
+        viewModel.getGroupInfo(groupId).subscribe(test)
+
+        /* Then */
+        test.assertValue { it == group }
+    }
+
+    @Test
+    fun `Given a group Id - When call get member list  - Then return list users `() {
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/group/info/GroupInfoViewModelTest.kt]: `Id` change to `id` also check all this class for this case

**Comment by at-dev-support-01** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">All test passed. (26/26)</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: app/src/test/java/vn/asiantech/way/group/info/GroupInfoViewModelTest.kt [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+        val test = TestObserver<Boolean>()
+        val groupId = "groupId"
+        val newOwnerId = "newOwnerId"
+        `when`(groupRepository.changeGroupOwner(TestUtil.any(), TestUtil.any())).thenReturn(Single.just(true))
+
+        /* When */
+        viewModel.changeGroupOwner(groupId, newOwnerId).subscribe(test)
+
+        /* Then */
+        test.assertValue(true)
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/group/info/GroupInfoViewModelTest.kt]: change this line to `test.assertValue(it)`

File: app/src/test/java/vn/asiantech/way/group/info/GroupInfoViewModelTest.kt [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        val test = TestObserver<Boolean>()
+        val groupId = "groupId"
+        val newOwnerId = "newOwnerId"
+        `when`(groupRepository.changeGroupOwner(TestUtil.any(), TestUtil.any())).thenReturn(Single.just(false))
+
+        /* When */
+        viewModel.changeGroupOwner(groupId, newOwnerId).subscribe(test)
+
+        /* Then */
+        test.assertValue(false)
```
Comment (at-vinhhuynh) [inline app/src/test/java/vn/asiantech/way/group/info/GroupInfoViewModelTest.kt]: change this line to `test.assertValue(!it)`

**Comment by at-vinhhuynh** [comment]
@at-tienhoang 
I finish review this PR, please check it again.
thank you!

File: app/src/test/java/vn/asiantech/way/group/info/GroupInfoViewModelTest.kt [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+    fun `Given a group id and new owner id - When call change group owner  - Then return false `() {
+        /* Given */
+        val test = TestObserver<Boolean>()
+        `when`(groupRepository.changeGroupOwner(TestUtil.any(), TestUtil.any())).thenReturn(Single.just(false))
+
+        /* When */
+        viewModel.changeGroupOwner("", "").subscribe(test)
+
+        /* Then */
+        test.assertValue { !it }
```
Comment (at-tienhoang) [inline app/src/test/java/vn/asiantech/way/group/info/GroupInfoViewModelTest.kt]: Pease change to `it == false`

File: app/src/test/java/vn/asiantech/way/group/info/GroupInfoViewModelTest.kt [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    fun `Given a group id and new owner id - When call change group owner  - Then return true `() {
+        /* Given */
+        val test = TestObserver<Boolean>()
+        `when`(groupRepository.changeGroupOwner(TestUtil.any(), TestUtil.any())).thenReturn(Single.just(true))
+
+        /* When */
+        viewModel.changeGroupOwner("", "").subscribe(test)
+
+        /* Then */
+        test.assertValue { it }
```
Comment (at-tienhoang) [inline app/src/test/java/vn/asiantech/way/group/info/GroupInfoViewModelTest.kt]: change to `it == true`

**Pull request merged**